server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/DLPConfigurationRoutesTest.java

```diff
@@ -0,0 +1,575 @@
+import org.apache.james.dlp.api.DLPConfigurationStore;
+import org.apache.james.dlp.eventsourcing.EventSourcingDLPConfigurationStore;
+import org.apache.james.dnsservice.api.DNSService;
+import org.apache.james.domainlist.api.DomainList;
+import org.apache.james.domainlist.memory.MemoryDomainList;
+import org.apache.james.eventsourcing.eventstore.memory.InMemoryEventStore;
+import org.apache.james.metrics.logger.DefaultMetricFactory;
+import org.apache.james.webadmin.WebAdminServer;
+import org.apache.james.webadmin.WebAdminUtils;
+import org.apache.james.webadmin.utils.JsonTransformer;
+import org.eclipse.jetty.http.HttpStatus;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import com.google.common.collect.ImmutableList;
+import com.jayway.restassured.RestAssured;
+import com.jayway.restassured.builder.RequestSpecBuilder;
+import com.jayway.restassured.http.ContentType;
+import com.jayway.restassured.specification.RequestSpecification;
+
+class DLPConfigurationRoutesTest {
+
+    private static final String DEFAULT_DOMAIN = "james.org";
+    private static final Domain SENDER_DOMAIN = Domain.of(DEFAULT_DOMAIN);
+    private static final String DOMAIN_2 = "apache.org";
+    private static final Domain SENDER_DOMAIN_2 = Domain.of(DOMAIN_2);
+
+    private static final DLPConfigurationItem CONFIGURATION_ITEM_1 = DLPConfigurationItem.builder()
+        .id(DLPConfigurationItem.Id.of("1"))
+        .explanation("explanation 1")
+        .expression(DEFAULT_DOMAIN)
+        .targetsSender()
+        .targetsRecipients()
+        .targetsContent()
+        .build();
+    private static final DLPConfigurationItem CONFIGURATION_ITEM_2 = DLPConfigurationItem.builder()
+        .id(DLPConfigurationItem.Id.of("2"))
+        .expression(DEFAULT_DOMAIN)
+        .targetsSender()
+        .build();
+
+    private static final List<DLPConfigurationItem> CONFIGURATION_ITEMS_FOR_DOMAIN_2 = ImmutableList.of(DLPConfigurationItem.builder()
+        .id(DLPConfigurationItem.Id.of("3"))
+        .expression(DOMAIN_2)
+        .targetsSender()
+        .build());
+
+    private static final List<DLPConfigurationItem> CONFIGURATION_ITEMS = ImmutableList.of(CONFIGURATION_ITEM_1, CONFIGURATION_ITEM_2);
+
+    private WebAdminServer webAdminServer;
+    private EventSourcingDLPConfigurationStore dlpStore;
+
+    private void createServer(DLPConfigurationStore dlpConfigurationStore, DomainList domainList) throws Exception {
+        webAdminServer = WebAdminUtils.createWebAdminServer(
+            new DefaultMetricFactory(),
+            new DLPConfigurationRoutes(dlpConfigurationStore, domainList, new JsonTransformer()));
+        webAdminServer.configure(NO_CONFIGURATION);
+        webAdminServer.await();
+
+        requestSpecification = buildRequestSpecification(webAdminServer);
+    }
+
+    RequestSpecification buildRequestSpecification(WebAdminServer server) {
```

Comment:
WebAdminUtils offers a buildRequestSpecification to reduce the boiler plate. Please use it.